Cole/mask to gan

zimmerman-cole opened the pull request:
Theoretically, the final version of our code - except for this tensorflow error that's happening on line 202 of tsnake/main.py

zimmerman-cole committed:
Fix `threshold` bug

- In Allen's conversion to Eq (6), the argument `threshold` was reused as the parameter `k`, but since MaskedRegion.initialize_tsnake had a default `threshold=100` this meant that the default value of `k` in Eq (6) was 100.
- I renamed the argument `threshold` to `k` in utils.img_inflation_force() and MaskedRegion.initialize_tsnake(), so the bug should be fixed.

zimmerman-cole committed:
Update notebooks for `threshold` fix

zimmerman-cole committed:
Fix some bugs in jiahui/

zimmerman-cole committed:
Merge branch 'master' of https://github.com/CS269-Capstone/t-snake-mask-generation into cole/mask-to-gan

zimmerman-cole committed:
Code should now theoretically work

zimmerman-cole committed:
Miscellaneous bug fixes

zimmerman-cole committed:
Add notebooks/Full-Demonstration.ipynb

allen-n committed:
Fixed opencv package name.

allen-n commented:
Found the error, the flood fill is flooding to every single node, will try to fix it.

allen-n committed:
Fixed reference issues, flood fill now partially working.

allen-n committed:
Local file changes to make troubleshooting easier. Drop in squash.

allen-n commented:
Error was that I was accessing by temporary reference since I did for loops over containers instead of indicies. I.e. for a in interable rather than for i in range(iterable). What an error.

allen-n committed:
Incremental improvement. Grid object in main now correctly recognizes turned-on cells. However, number of turned on cells disagrees, AND almost every single cell is turned on.

allen-n committed:
Grid norm scale seems to have influence on number of cells added to grid-node-queue. Squash.

zimmerman-cole committed:
Remove requirement for old tensorflow versions

zimmerman-cole committed:
Remove now-broken inpainting code ;(

zimmerman-cole committed:
Change Main comparison to count # masked pixels

- Remove Main.compare_inpainted_images()
- Add Main.compare_masks(), which tracks the percent reduction in number of masked pixels instead of difference in inpainting quality

zimmerman-cole committed:
Update Full-Demonstration.ipynb

zimmerman-cole committed:
Fix coordinate-system issue in snakes->mask

- Remember that each snake operates within its respective MaskedRegion, and the coordinate system of that MaskedRegion is different from the image as a whole
- Updated the final portion of Main.run() to make the snake node coordinates consistent with the full image coordinates

zimmerman-cole committed:
Fix minor bugs in and augment Main.compare_masks()

- Percent reduction in # of masked pixels now displayed correctly
- Add method Main.process_snakes_for_mask_generation() which changes the snakes nodes' coordinates to be consistent with the full image coordinates
- Main.compare_snakes() now displays the snake(s) as well
